src/network/__init__.py

```diff
@@ -4,19 +4,19 @@
 
 try:
     from .announcethread import AnnounceThread
-    from .connectionpool import BMConnectionPool
+    from .connectionpool import BMConnectionPool as BMConnectionPool_cls
+    BMConnectionPool = BMConnectionPool_cls()
 except ImportError:
```

Comment:
I'm not sure this try/except is needed anymore once we fix the circular imports.